common/src/api/external/mod.rs

```diff
@@ -2824,7 +2805,7 @@ pub struct SwitchPortRouteConfig {
     pub dst: oxnet::IpNet,
 
     /// The route's gateway address.
-    pub gw: oxnet::IpNet,
```

Comment:
why can this go from `IpNet` to `IpAddr`?

Reply:
This was likely never supposed to be an `IpNet`. The user provides an `IpAddr` when they create the configuration:

https://github.com/oxidecomputer/omicron/blob/abc98f2575a0ac1ae99dc9c1922381350a62937e/nexus/types/src/external_api/params.rs#L1900

The DB stores singular ipv4 addresses as a full CIDR with /32, so that's probably how we ended up returning an `IpNet` instead of an `IpAddr` like what the user supplied.